src/controller/src/lib.rs

```diff
@@ -135,29 +135,56 @@ impl ConcreteComputeInstanceReplicaLocation {
     }
 }
 
-/// Logging configuration of a replica
+/// Logging configuration of a replica.
+/// Changing this type requires a catalog storage migration!
 #[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
 pub enum ConcreteComputeInstanceReplicaLogging {
     /// Instantiate default logging configuration upon system start.
-    /// To configure a replica without logging, Concrete(vec![]) should be used.
+    /// To configure a replica without logging, ConcreteViews(vec![],vec![]) should be used.
     Default,
-    /// Logging collections have already been built for this replica.
+    /// Logging sources have been built for this replica. Upon system restart,
+    /// this will be replaced with ConcreteViews.
     Concrete(Vec<(LogVariant, GlobalId)>),
+    /// Logging sources and views have been built for this replica.
+    ConcreteViews(Vec<(LogVariant, GlobalId)>, Vec<(LogView, GlobalId)>),
 }
 
 impl ConcreteComputeInstanceReplicaLogging {
-    /// Return all logs described
+    /// Return all log sources described
     pub fn get_logs(&self) -> Vec<(LogVariant, GlobalId)> {
         match self {
             ConcreteComputeInstanceReplicaLogging::Default => vec![],
             ConcreteComputeInstanceReplicaLogging::Concrete(logs) => logs.clone(),
+            ConcreteComputeInstanceReplicaLogging::ConcreteViews(logs, _) => logs.clone(),
         }
     }
 
+    /// Returns log ids described
+    pub fn get_views(&self) -> Vec<(LogView, GlobalId)> {
+        match self {
+            ConcreteComputeInstanceReplicaLogging::Default => vec![],
+            ConcreteComputeInstanceReplicaLogging::Concrete(_) => vec![],
+            ConcreteComputeInstanceReplicaLogging::ConcreteViews(_, views) => views.clone(),
+        }
+    }
+
+    /// Returns log ids described
```

Comment:
I think this comment should probably include that it's only views that are returned.